### PR TITLE
azurerm_virtual_network_gateway(_connection): Fix acctest failures

### PR DESCRIPTION
**Pull request by magodo**
Fix following acctest failures:

```bash
💢 TF_ACC=1 go test -v -timeout=3h ./azurerm/internal/services/network -run="TestAccVirtualNetworkGatewayConnection_useLocalAzureIpAddressEnabled|TestAccVirtualNetworkGateway_privateIpAddressEnabled|TestAccVirtualNetworkGateway_activeActive|TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth"
2021/05/17 14:43:45 [DEBUG] not using binary driver name, it's no longer needed
2021/05/17 14:43:45 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccVirtualNetworkGatewayConnection_useLocalAzureIpAddressEnabled
=== PAUSE TestAccVirtualNetworkGatewayConnection_useLocalAzureIpAddressEnabled
=== RUN   TestAccVirtualNetworkGateway_activeActive
=== PAUSE TestAccVirtualNetworkGateway_activeActive
=== RUN   TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth
=== PAUSE TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth
=== RUN   TestAccVirtualNetworkGateway_activeActiveEnableBgpWithAPIPA
=== PAUSE TestAccVirtualNetworkGateway_activeActiveEnableBgpWithAPIPA
=== RUN   TestAccVirtualNetworkGateway_privateIpAddressEnabled
=== PAUSE TestAccVirtualNetworkGateway_privateIpAddressEnabled
=== CONT  TestAccVirtualNetworkGatewayConnection_useLocalAzureIpAddressEnabled
=== CONT  TestAccVirtualNetworkGateway_activeActiveEnableBgpWithAPIPA
=== CONT  TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth
=== CONT  TestAccVirtualNetworkGateway_activeActive
=== CONT  TestAccVirtualNetworkGateway_privateIpAddressEnabled
--- PASS: TestAccVirtualNetworkGateway_activeActive (1565.88s)
--- PASS: TestAccVirtualNetworkGateway_activeActiveEnableBgpWithAPIPA (1769.03s)
--- PASS: TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth (1854.31s)
--- PASS: TestAccVirtualNetworkGateway_privateIpAddressEnabled (4550.10s)
--- PASS: TestAccVirtualNetworkGatewayConnection_useLocalAzureIpAddressEnabled (5030.25s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network     5030.423s
```

### CODE REVIEW

File: azurerm/internal/services/network/virtual_network_gateway_resource_test.go [hunk 1]
```diff
@@ -179,10 +179,6 @@ func TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth(t *testing.T) {
 			Config: r.vpnClientConfigAzureAdAuth(data),
 			Check: resource.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
-				check.That(data.ResourceName).Key("vpn_client_configuration.0.aad_tenant").HasValue(fmt.Sprintf("https://login.microsoftonline.com/%s/", data.RandomString)),
-				check.That(data.ResourceName).Key("vpn_client_configuration.0.aad_audience").HasValue("41b23e61-6c1e-4545-b367-cd054e0ed4b4"),
-				check.That(data.ResourceName).Key("vpn_client_configuration.0.aad_issuer").HasValue(fmt.Sprintf("https://sts.windows.net/%s/", data.RandomString)),
-				check.That(data.ResourceName).Key("vpn_client_configuration.0.vpn_client_protocols.#").HasValue("1"),
```
Comment (manicminer) [inline azurerm/internal/services/network/virtual_network_gateway_resource_test.go:185]: If we are removing these checks, we should add an import step instead. Or we could fix the checks by specifying the correct tenant ID etc

**Review by manicminer — CHANGES_REQUESTED** [review]
Thanks @magodo, this mostly LGTM, I just have a query about the checks being removed.

File: azurerm/internal/services/network/virtual_network_gateway_resource_test.go (hunk 1 above)
Comment (magodo) [reply azurerm/internal/services/network/virtual_network_gateway_resource_test.go:185]: @manicminer Sorry about this, I forgot to check this. The test passed after adding the ImportStep:

```bash
💢 TF_ACC=1 go test -v -timeout=3h ./azurerm/internal/services/network -run="TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth"
2021/05/18 09:19:25 [DEBUG] not using binary driver name, it's no longer needed
2021/05/18 09:19:25 [DEBUG] not using binary driver name, it's no longer needed
=== RUN   TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth
=== PAUSE TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth
=== CONT  TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth
--- PASS: TestAccVirtualNetworkGateway_vpnClientConfigAzureAdAuth (1677.17s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/network     1677.380s
```

**Review by katbyte — APPROVED** [review]
LGTM 👍

**Pull request merged**